persistent/Database/Persist/Types/Base.hs

```diff
@@ -131,6 +131,8 @@ data EntityDef = EntityDef
     -- ^ The name of the entity as Haskell understands it.
     , entityDB      :: !EntityNameDB
     -- ^ The name of the database table corresponding to the entity.
+    , entitySchema  :: !(Maybe Text)
+    -- ^ The schema name of the database table.
```

Comment:
Let's make another `newtype SchemaNameDB` for this - will make it easier to disambiguate what is and isn't a schema vs any other bit of text